### PR TITLE
Feature/visualize refactor

### PR DESCRIPTION
**Pull request by Jammy2211**
Cleans up visualization code to make default images appear closer to publication quality.

The `subplot_fit` we use to inspect model-fits now have 9 panels, so we do not have to alternative between `subplot_of_planes`:

image

### CODE REVIEW

File: autogalaxy/interferometer/model/analysis.py [hunk 1]
```diff
@@ -330,6 +305,34 @@ def fit_interferometer_via_plane_from(
...
+        ----------
+        paths
+            The PyAutoFit paths object which manages all paths, e.g. where the non-linear search outputs are stored,
+            visualization and the pickled objects used by the aggregator output by this function.
+        model
+            The PyAutoFit model object, which includes model components representing the galaxies that are fitted to
+            the imaging data.
+        """
+
+        if paths.is_complete or not os.environ.get("PYAUTOFIT_TEST_MODE") == "1":
```
Comment (rhayes777) [inline autogalaxy/interferometer/model/analysis.py:324]: So if it's not test mode we never want to visualise before a fit? That doesn't seem right?

File: autogalaxy/interferometer/model/result.py [hunk 2]
```diff
@@ -108,36 +104,36 @@ def visibilities_galaxy_dict(self) -> {str: Galaxy}:
...
         """
 
-        hyper_galaxy_visibilities_path_dict = {}
+        adapt_galaxy_visibilities_path_dict = {}
 
         for path, galaxy in self.path_galaxy_tuples:
-            hyper_galaxy_visibilities_path_dict[path] = self.visibilities_galaxy_dict[
+            adapt_galaxy_visibilities_path_dict[path] = self.visibilities_galaxy_dict[
                 path
             ]
```
Comment (rhayes777) [inline autogalaxy/interferometer/model/result.py:120]: Could be a dict comprehension

File: autogalaxy/plane/plane.py [hunk 3]
```diff
@@ -100,6 +98,13 @@ def cls_list_from(self, cls: Type) -> List:
     def galaxies_with_cls_list_from(self, cls: Type) -> List[Galaxy]:
         return list(filter(lambda galaxy: galaxy.has(cls=cls), self.galaxies))
 
+    @property
+    def adapt_galaxies_with_pixelization_image_list(self) -> List[aa.Array2D]:
```
Comment (rhayes777) [inline autogalaxy/plane/plane.py:102]: Docs?

File: autogalaxy/plot/include/two_d.py [hunk 4]
```diff
@@ -53,12 +57,24 @@ def mass_profile_centres(self):
...
+
+    @property
+    def radial_critical_curves(self):
+        return self.load(
+            value=self._radial_critical_curves, name="radial_critical_curves"
+        )
+
+    @property
+    def tangential_caustics(self):
+        return self.load(value=self._tangential_caustics, name="tangential_caustics")
```
Comment (rhayes777) [inline autogalaxy/plot/include/two_d.py:73]: Docs?

File: autogalaxy/profiles/mass/multipole/power_law_m4.py [hunk 5]
```diff
@@ -0,0 +1,253 @@
...
+
+        grid=al.Grid2D.uniform(shape_native=(10, 10), pixel_scales=0.1)
+
+        deflections = galaxy.deflections_yx_2d_from(
+            grid=grid
+        )
+        """
+        super().__init__(centre=centre, ell_comps=(0.0, 0.0))
+
+        self.m = 4
```
Comment (rhayes777) [inline autogalaxy/profiles/mass/multipole/power_law_m4.py:145]: Where does this number come from?

File: test_autogalaxy/legacy/test_setup.py [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+
+    setup = ag.legacy.SetupAdapt(search_noise_cls=None, search_noise_dict=None)
+    assert setup.search_noise_cls == af.DynestyStatic
+    assert setup.search_noise_dict == {"nlive": 50, "sample": "rwalk"}
+
+    setup = ag.legacy.SetupAdapt(
+        search_noise_cls=af.DynestyDynamic, search_noise_dict={"hello": "there"}
+    )
+    assert setup.search_noise_cls == af.DynestyDynamic
+    assert setup.search_noise_dict == {"hello": "there"}
```
Comment (rhayes777) [inline test_autogalaxy/legacy/test_setup.py:29]: Should be four separate tests

**Review by rhayes777 — APPROVED** [review]

File: autogalaxy/plane/plane.py (hunk 3 above)
Comment (Jammy2211) [reply autogalaxy/plane/plane.py:102]: One day.

**Pull request merged**